noria-server/mir/src/optimize.rs

```diff
@@ -1,19 +1,217 @@
 use dataflow::ops::filter::FilterCondition;
+use dataflow::ops::grouped::filteraggregate::FilterAggregation;
+use dataflow::ops::grouped::aggregate::Aggregation;
 use node::{MirNode, MirNodeType};
 use query::MirQuery;
 use MirNodeRef;
 
 use std::collections::HashMap;
 
-pub fn optimize(q: MirQuery) -> MirQuery {
+pub fn optimize(mut q: &mut MirQuery) -> Vec<MirNodeRef> {
```

Comment:
It'd be good to document what this returns.

Reply:
Agreed. Done.